Common/cpp/SharedItems/Shareable.cpp

```diff
@@ -15,6 +29,29 @@ void ShareableValue::adaptCache(jsi::Runtime &rt, const jsi::Value &value) {
 }
 
 void ShareableValue::adapt(jsi::Runtime &rt, const jsi::Value &value, ValueType objectType) {
+  if (value.isObject()) {
+    jsi::Object object = value.asObject(rt);
+    
+    if (object.hasProperty(rt, HIDDEN_PROPERTY_NAME.c_str()) and
```

Comment:
we don't need hasProperty check. See below where we test for existence of `__worklet` property